packages/gotrue/lib/src/gotrue_client.dart

```diff
@@ -773,29 +773,14 @@ class GoTrueClient {
     );
   }
 
-  /// Recover session from persisted session json string.
-  /// Persisted session json has the format { currentSession, expiresAt }
-  ///
-  /// currentSession: session json object, expiresAt: timestamp in seconds
+  /// Recover session from stringified [Session].
   Future<AuthResponse> recoverSession(String jsonStr) async {
-    final persistedData = json.decode(jsonStr) as Map<String, dynamic>;
-    final currentSession =
-        persistedData['currentSession'] as Map<String, dynamic>?;
-    final expiresAt = persistedData['expiresAt'] as int?;
-    if (currentSession == null) {
-      throw _notifyException(AuthException('Missing currentSession.'));
-    }
-    if (expiresAt == null) {
-      throw _notifyException(AuthException('Missing expiresAt.'));
-    }
-
-    final session = Session.fromJson(currentSession);
+    final session = Session.fromJson(json.decode(jsonStr));
```

Comment:
Is the `jsonStr` long? We may take advantage of the `compute` function if it's a complicated parsing

Reply:
When auto formatted, it looks like the following. I don't think it's necessary to spin up an isolate for this.
```
{
    "access_token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2OTM5MDk4MzQsInN1YiI6IjE4YmM3YTRlLWMwOTUtNDU3My05M2RjLWUwYmUyOWJhZGE5NyIsImVtYWlsIjoiZmFrZTFAZW1haWwuY29tIiwicGhvbmUiOiIxNjY2MDAwMDAwMDAiLCJhcHBfbWV0YWRhdGEiOnsicHJvdmlkZXIiOiJlbWFpbCIsInByb3ZpZGVycyI6WyJlbWFpbCJdfSwidXNlcl9tZXRhZGF0YSI6e30sInJvbGUiOiIiLCJhYWwiOiJhYWwxIiwiYW1yIjpbeyJtZXRob2QiOiJwYXNzd29yZCIsInRpbWVzdGFtcCI6MTY5MzkwNjIzNH1dLCJzZXNzaW9uX2lkIjoiZDdhZDdhMzgtY2YzOS00ZGI3LTllNDYtMTlmNDc2ZjViNDBmIn0.ZtPq2EYKIvH8fyjJ7sbdtLPjvzS1DQFO4a1UHkojrkU",
    "expires_in": 3600,
    "refresh_token": "a-x22GFhn19riGhRm52kZg",
    "token_type": "bearer",
    "provider_token": null,
    "provider_refresh_token": null,
    "user": {
        "id": "18bc7a4e-c095-4573-93dc-e0be29bada97",
        "app_metadata": {
            "provider": "email",
            "providers": [
                "email"
            ]
        },
        "user_metadata": {},
        "aud": "",
        "confirmation_sent_at": null,
        "recovery_sent_at": null,
        "email_change_sent_at": null,
        "new_email": null,
        "invited_at": null,
        "action_link": null,
        "email": "fake1@email.com",
        "phone": "166600000000",
        "created_at": "2023-09-05T09:30:34.682647Z",
        "confirmed_at": "2023-09-05T09:30:34.682647Z",
        "email_confirmed_at": "2023-09-05T09:30:34.682647Z",
        "phone_confirmed_at": "2023-09-05T09:30:34.682647Z",
        "last_sign_in_at": "2023-09-05T09:30:34.860853363Z",
        "role": "",
        "updated_at": "2023-09-05T09:30:34.863812Z",
        "identities": [
            {
                "id": "18bc7a4e-c095-4573-93dc-e0be29bada97",
                "user_id": "18bc7a4e-c095-4573-93dc-e0be29bada97",
                "identity_data": {
                    "email": "fake1@email.com",
                    "sub": "18bc7a4e-c095-4573-93dc-e0be29bada97"
                },
                "provider": "email",
                "created_at": "2023-09-05T09:30:34.682647Z",
                "last_sign_in_at": "2023-09-05T09:30:34.682647Z",
                "updated_at": "2023-09-05T09:30:34.682647Z"
            }
        ],
        "factors": [
            {
                "id": "1d3aa138-da96-4aea-8217-af07daa6b82d",
                "friendly_name": "UnverifiedFactor",
                "factor_type": "totp",
                "status": "unverified",
                "created_at": "2023-09-05T09:30:34.682647Z",
                "updated_at": "2023-09-05T09:30:34.682647Z"
            }
        ]
    }
}
```